src/sfctl/tests/help_text_test.py

```diff
@@ -0,0 +1,364 @@
+                get     : Gets something.
+                events  : Gets the next segment of the Chaos events...
+                      range.
+
+        or
+
+        2.  Command
+                sfctl application create: Creates a Service Fabric appli...
+
+            Arguments
+                --app-version [Required]: The version of the application type as defined in the
+                                          manifest.
+                --timeout -t            : Server timeout in seconds.  Default: 60.
+
+            Global Arguments
+                --debug                 : Increase logging verbosity to show all debug logs.
+        """
+
+        try:
+            # This variable tracks what sections of the help text we are in
+            # Possibilities are Group, Subgroups, Commands, Command, Arguments,
+            # and Global Arguments.
+            # Once we no longer support python 2, change section options of enums
+            section = 'Start'
+
+            # A tracker to know which subgroup or command has appeared in help text
+            subgroups_index = 0
+            commands_index = 0
+
+            help_command = command_input + ' -h'
+
+            # Call the provided command in command line
+            # Do not split the help_command, as that breaks behavior: Linux ignores the splits.
+            pipe = Popen(help_command, shell=True, stdout=PIPE, stderr=PIPE)
+            # returned_string and err are returned as bytes
+            (returned_string, err) = pipe.communicate()
+
+            # If there is an error message, then assert
+            if err:
+                err = err.decode('utf-8')
+                print(err, file=stderr)
+                self.assertEqual(b'', err, msg='ERROR: in command: ' + help_command)
+
+            # If there is no help text, then assert
+            if not returned_string:
+                self.fail('No help text in command: ' + help_command)
+
+            returned_string = returned_string.decode('utf-8')
+            lines = returned_string.splitlines()
+
+            for line in lines:
+
+                print(line, file=stderr)
+
+                # Skip empty lines
+                if not line.strip():
+                    continue
+
+                # Check if we want to mark the start of a new section
+                # Check this by seeing if the line is a top level description, ie: 'Commands:'
+                # These are characterized by a new line with text starting without spaces
+                read_section_output = self._validate_output_read_section_name(line)
+                if read_section_output is not None:
+                    section = read_section_output
+                    continue
+
+                    # If this line is a section designator, no additional processing
+                    # is required. Move on to the next line.
+
+                updated_indices = self._validate_output_validate_line(command_input, line, section,
```

Comment:
I'd break up the validate command into a command that actually performs the validation and another command that can be used to determine what other subcommands exist, that way you don't have this weird return type.